lib/api/CMakeLists.txt

```diff
@@ -8,10 +8,18 @@ endif()
 
 # Helper function to construct commands and dependencies.
 function(wasmedge_add_static_lib_component_command target)
-  list(APPEND CMDS
-    COMMAND ${CMAKE_COMMAND} -E make_directory objs/${target}
-    COMMAND ${CMAKE_AR} -x $<TARGET_FILE:${target}> --output=objs/${target}
-  )
+  if(APPLE) 
+    list(APPEND CMDS
+      COMMAND ${CMAKE_COMMAND} -E make_directory objs/${target}
+      COMMAND ${CMAKE_AR} -x $<TARGET_FILE:${target}>
+      COMMAND ${CMAKE_AR} -t $<TARGET_FILE:${target}> | xargs -I '{}' mv '{}' objs/${target}
+    )
+  else()
+    list(APPEND CMDS
+      COMMAND ${CMAKE_COMMAND} -E make_directory objs/${target}
+      COMMAND ${CMAKE_AR} -x $<TARGET_FILE:${target}> --output=objs/${target}
+    )
+  endif()
```

Comment:
Can you also implement this in the next function `wasmedge_add_libs_component_command`?
Thanks.

Reply:
Sure, I forget to check other functions